server/api/src/routes/error.rs

```diff
@@ -0,0 +1,180 @@
+    fn from(e: ActivateAccountError) -> Self {
+        match e {
+            ActivateAccountError::InvalidToken => Self::new(StatusCode::BAD_REQUEST, e.to_string()),
+            ActivateAccountError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
+
+impl<D: Serialize> From<FindExpensesWithTransactionsByUserIdError> for ErrorMsg<D> {
+    fn from(e: FindExpensesWithTransactionsByUserIdError) -> Self {
+        match e {
+            FindExpensesWithTransactionsByUserIdError::UnauthorizedUser(_) => {
+                Self::new(StatusCode::FORBIDDEN, e.to_string())
+            }
+            FindExpensesWithTransactionsByUserIdError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
+
+impl<D: Serialize> From<CreateExpenseError> for ErrorMsg<D> {
+    fn from(e: CreateExpenseError) -> Self {
+        match e {
+            CreateExpenseError::InvalidPredefinedExpense
+            | CreateExpenseError::InvalidRelatedType(_) => {
+                Self::new(StatusCode::NOT_FOUND, e.to_string())
+            }
+            CreateExpenseError::InvalidStartDate(_) => {
+                Self::new(StatusCode::BAD_REQUEST, e.to_string())
+            }
+            CreateExpenseError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
+
+impl<D: Serialize> From<CreatePredefinedExpenseError> for ErrorMsg<D> {
+    fn from(e: CreatePredefinedExpenseError) -> Self {
+        match e {
+            CreatePredefinedExpenseError::InvalidRelatedType(_) => {
+                Self::new(StatusCode::NOT_FOUND, e.to_string())
+            }
+            CreatePredefinedExpenseError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
+
+impl<D: Serialize> From<CreateTransactionError> for ErrorMsg<D> {
+    fn from(e: CreateTransactionError) -> Self {
+        match e {
+            CreateTransactionError::InvalidExpenseId
+            | CreateTransactionError::InvalidCurrencyType => {
+                Self::new(StatusCode::NOT_FOUND, e.to_string())
+            }
+            CreateTransactionError::UserUnauthorized(_) => {
+                Self::new(StatusCode::UNAUTHORIZED, e.to_string())
+            }
+            CreateTransactionError::InvalidStartDate(_) => {
+                Self::new(StatusCode::BAD_REQUEST, e.to_string())
+            }
+            CreateTransactionError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
+
+impl<D: Serialize> From<DeleteTransactionByIdError> for ErrorMsg<D> {
+    fn from(e: DeleteTransactionByIdError) -> Self {
+        match e {
+            DeleteTransactionByIdError::InvalidTransaction => {
+                Self::new(StatusCode::NOT_FOUND, e.to_string())
+            }
+            DeleteTransactionByIdError::UserUnauthorized(_) => {
+                Self::new(StatusCode::UNAUTHORIZED, e.to_string())
+            }
+            DeleteTransactionByIdError::DatabaseError(db_error) => db_error.into(),
+        }
+    }
+}
```

Comment:
I guess this giant is the one, that can be refactored with New type pattern

Reply:
Nope, newtype would be used instead of the validation library as a way to make sure that Strings for example that have been validated once, does not get corrupted somewhere in the middle because of developer error, and then saved.
If you have a suggestion tough to make this better feel free to say it, I don't really have any ideas.

Reply:
I commented for this explanation, thanks. 

The improvement of this will be on another day.